brainscore_vision/model_helpers/activations/core.py

```diff
@@ -53,38 +97,46 @@ def from_stimulus_set(self, stimulus_set, layers, stimuli_identifier=None):
-        activations = attach_stimulus_set_meta(activations, stimulus_set)
+        activations = attach_stimulus_set_meta(activations, stimulus_set, number_of_trials=self.number_of_trials,
+                                               require_variance=require_variance)
         return activations
 
-    def from_paths(self, stimuli_paths, layers, stimuli_identifier=None):
+    def from_paths(self, stimuli_paths, layers, stimuli_identifier=None, require_variance=None):
         if layers is None:
             layers = ['logits']
         if self.identifier and stimuli_identifier:
             fnc = functools.partial(self._from_paths_stored,
-                                    identifier=self.identifier, stimuli_identifier=stimuli_identifier)
+                                    identifier=self.identifier,
+                                    stimuli_identifier=stimuli_identifier,
+                                    require_variance=require_variance)
         else:
             self._logger.debug(f"self.identifier `{self.identifier}` or stimuli_identifier {stimuli_identifier} "
                                f"are not set, will not store")
             fnc = self._from_paths
-        # In case stimuli paths are duplicates (e.g. multiple trials), we first reduce them to only the paths that need
-        # to be run individually, compute activations for those, and then expand the activations to all paths again.
-        # This is done here, before storing, so that we only store the reduced activations.
-        reduced_paths = self._reduce_paths(stimuli_paths)
-        activations = fnc(layers=layers, stimuli_paths=reduced_paths)
-        activations = self._expand_paths(activations, original_paths=stimuli_paths)
+        if require_variance:
+            activations = fnc(layers=layers, stimuli_paths=stimuli_paths, require_variance=require_variance)
+        else:
+            # In case stimuli paths are duplicates (e.g. multiple trials), we first reduce them to only the paths that need
+            # to be run individually, compute activations for those, and then expand the activations to all paths again.
+            # This is done here, before storing, so that we only store the reduced activations.
+            reduced_paths = self._reduce_paths(stimuli_paths)
+            activations = fnc(layers=layers, stimuli_paths=reduced_paths, require_variance=require_variance)
+            activations = self._expand_paths(activations, original_paths=stimuli_paths)
         return activations
 
     @store_xarray(identifier_ignore=['stimuli_paths', 'layers'], combine_fields={'layers': 'layer'})
-    def _from_paths_stored(self, identifier, layers, stimuli_identifier, stimuli_paths):
+    def _from_paths_stored(self, identifier, layers, stimuli_identifier,
+                           stimuli_paths, number_of_trials: int = 1, require_variance: bool = False):
         return self._from_paths(layers=layers, stimuli_paths=stimuli_paths)
```

Comment:
```suggestion
        return self._from_paths(layers=layers, stimuli_paths=stimuli_paths, require_variance=require_variance)
```

Reply:
I believe that we explicitly should not do this, right? (similarly to number_of_trials), as if the data are stored, microsaccades will exist in those stored arrays already

Reply:
the `stored` in the function name just refers to the `@store_xarray` decorator which will still run the function if no result has been stored.